components/eamxx/src/diagnostics/aerocom_cld.cpp

```diff
@@ -45,6 +45,10 @@ void AeroComCld::set_grids(
   EKAT_REQUIRE_MSG(m_index_map.size() == m_units_map.size(),
                    "Error! Some inconsistency in AeroComCld: index and units "
                    "maps do not match!\n");
+  // Ensure m_index_map and m_ndiag match
+  EKAT_REQUIRE_MSG(static_cast<int>(m_index_map.size()) == m_ndiag,
```

Comment:
Why static_cast here? You could `(int)m_index_map.size()` instead.

Reply:
Never mind, looking into it more, `static_cast` is probably the better option.

Reply:
Yeah, static_cast will ensure the types are compatible, whereas I don't think `(int)` has any sort of checks. It shouldn't matter here, but `static_cast` is safer.